discord_bots/checks.py

```diff
@@ -90,23 +108,50 @@ async def is_admin_app_command(interaction: Interaction) -> bool:
             return False
 
 
-async def economy_enabled(interaction: Interaction) -> bool:
+async def is_command_channel(interaction: Interaction) -> bool:
     """
-    Check to wrap functions that require player economy to be enabled
+    Check that interactions are performed from the command channel
     """
-
-    if not interaction:
-        return False
-
-    if not ECONOMY_ENABLED:
-        await interaction.response.send_message(
-            "Player economy is disabled",
-            ephemeral=True
-        )
+    if not interaction.channel:
```

Comment:
Separating the conditional clauses removed a typing error that was present.
As if you try to interaction with `interaction.channel` before determining that it is `not None`, it presents an error.

Having a code path that returns out to False before checking if the `interaction.channel.id == CHANNEL_ID` removes the error from presenting.

Realistically, yes it could likely be compressed a bit, but I figured going for syntactic accuracy was preferable.

Reply:
Maybe the typechecker can't narrow it but the `not interaction.channel` being first in the condition should resolve that issue since it short-circuits?

Reply:
You make a fair point.
I've simplified the check in [acb7a20](https://github.com/tribesonecommunity/discord-bots/pull/93/commits/acb7a204e4d05402c361bc73c41b4638d6e35efa), and tested it as successful.